services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,189 @@
+@SdkPublicApi
+public final class CloudFrontUtilities {
+
+    private CloudFrontUtilities() {
+    }
+
+    // Used by low-level client
+    @SdkInternalApi
+    static CloudFrontUtilities create() {
+        return new CloudFrontUtilities();
+    }
+
+    /**
+     * Returns a signed URL with a canned policy that grants universal access to
+     * private content until a given date.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-canned-policy.html"
+     * >Creating a signed URL using a canned policy</a>.
+     *
+     * @param request
+     *            A CloudFrontSignerRequest configured with the following values:
+     *            resourceUrl, privateKey, keyPairId, expirationDate
+     * @return A signed URL that will permit access to a specific distribution
+     *         and S3 object.
+     */
+    public SignedUrl getSignedUrlWithCannedPolicy(CannedSignerRequest request) {
+        try {
+            String resourceUrl = request.resourceUrl();
+            String cannedPolicy = SigningUtils.buildCannedPolicy(resourceUrl, request.expirationDate());
+            byte[] signatureBytes = SigningUtils.signWithSha1Rsa(cannedPolicy.getBytes(UTF_8), request.privateKey());
+            String urlSafeSignature = SigningUtils.makeBytesUrlSafe(signatureBytes);
+            URI uri = URI.create(resourceUrl);
+            String encodedPath = uri.getPath()
+                   + (request.resourceUrl().indexOf('?') >= 0 ? "&" : "?")
+                   + "Expires=" + request.expirationDate().getEpochSecond()
+                   + "&Signature=" + urlSafeSignature
+                   + "&Key-Pair-Id=" + request.keyPairId();
+            return DefaultSignedUrl.builder().protocol(uri.getScheme()).domain(uri.getHost()).encodedPath(encodedPath).build();
+        } catch (InvalidKeyException e) {
+            throw SdkClientException.create("Could not sign url", e);
+        }
+    }
+
+    /**
+     * Returns a signed URL that provides tailored access to private content
+     * based on an access time window and an ip range. The custom policy itself
+     * is included as part of the signed URL (For a signed URL with canned
+     * policy, there is no policy included in the signed URL).
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-custom-policy.html"
+     * >Creating a signed URL using a custom policy</a>.
+     *
+     * @param request
+     *            A CloudFrontSignerRequest configured with the following values:
+     *            resourceUrl, privateKey, keyPairId, expirationDate, activeDate, ipRange
+     * @return A signed URL that will permit access to distribution and S3
+     *         objects as specified in the policy document.
+     */
+    public SignedUrl getSignedUrlWithCustomPolicy(CustomSignerRequest request) {
```

Comment:
Might be worth adding a `Consumer<CustomerSignerRequest.Builder>` overload as well. Our usability tests show that people like them: https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/java/software/amazon/awssdk/services/s3/S3Utilities.java#L205

Reply:
Added overloading methods that takes Consumer to all 4 methods